### PR TITLE
Instantiating CoreLoggerBase.Default in TokenCache

### PR DESCRIPTION
**Pull request by trwalke**
ADAL crashes when clearing the cache before attempting to acquire a token. This is because the CoreLoggerBase.Default is only instantiated during acquiretokenasync calls and the TokenCache has the ability to clear its cache and log it before these calls are made. This proposal involves instantiating CoreLoggerBase.Default on the construction of the ADAL TokenCache to avoid null references when logging from the TokenCache.

This was tested by following the reproduction steps outlined in #936
https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/936

### CODE REVIEW

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs [hunk 1]
```diff
@@ -85,6 +90,8 @@ public TokenCache(byte[] state)
             : this()
         {
             this.Deserialize(state);
+            if (CoreLoggerBase.Default == null)
+                CoreLoggerBase.Default = new AdalLogger(Guid.Empty);
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs]: not needed

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs [hunk 2]
```diff
@@ -76,6 +79,8 @@ static TokenCache()
         public TokenCache()
         {
             this.tokenCacheDictionary = new ConcurrentDictionary<AdalTokenCacheKey, AdalResultWrapper>();
+            if (CoreLoggerBase.Default == null)
+                CoreLoggerBase.Default = new AdalLogger(Guid.Empty);
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs]: should not be needed?

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs [hunk 3]
```diff
@@ -68,6 +68,9 @@ static TokenCache()
                 BeforeAccess = StorageDelegates.BeforeAccess,
                 AfterAccess = StorageDelegates.AfterAccess
             };
+
+            if (CoreLoggerBase.Default == null)
+                CoreLoggerBase.Default = new AdalLogger(Guid.Empty);
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs]: should be enough?

**Comment by henrik-me** [comment]
unit tests? other automated tests?

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs (hunk 3 above)
Comment (trwalke) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs]: @henrik-me For the Automation, yes, this is enough. The others were just to ensure all bases are covered.

**Pull request merged**